src/RecyclableMemoryStreamManager.cs

```diff
@@ -538,53 +502,74 @@ internal void ReturnBlocks(ICollection<byte[]> blocks, string tag)
 
         internal void ReportBlockCreated()
         {
-            this.BlockCreated?.Invoke();
+            Events.Writer.MemoryStreamNewBlockCreated(this.smallPoolInUseSize);
+            this.BlockCreated?.Invoke(new BlockCreatedEventArgs(this.smallPoolInUseSize));
         }
 
-        internal void ReportBlockDiscarded()
+        internal void ReportLargeBufferCreated(long requiredSize, bool pooled, Guid id, string tag, string callStack)
         {
-            this.BlockDiscarded?.Invoke();
+            if (pooled)
+            {
+                Events.Writer.MemoryStreamNewLargeBufferCreated(requiredSize, this.LargePoolInUseSize);
```

Comment:
id + tag?

Reply:
they are both there already, but I didn't normalize their order in the param list like I did elsewhere. Fixing.